src/cmd/binary.js

```diff
@@ -38,30 +38,73 @@ const DEFAULT_PRODUCT_ID = 65535;
 const DEFAULT_PRODUCT_VERSION = 65535;
 
 class BinaryCommand {
-	inspectBinary(binaryFile){
-		return Promise.resolve().then(() => {
-			if (!fs.existsSync(binaryFile)){
-				throw new VError(`Binary file not found ${binaryFile}`);
-			}
+	async inspectBinary(file) {
+		await this._checkFile(file);
+		const extractedFiles = await this._extractFiles(file);
+		const parsedBinaryInfo = await this._parseApplicationBinary(extractedFiles.application);
+		await this._verifyBundle(parsedBinaryInfo, extractedFiles.assets);
+	}
 
-			const parser = new Parser();
+	async _checkFile(file) {
+		// TODO: what happens if this is removed? What kind of error do we get?
```

Comment:
Can you check what happens without the file existance check for binary inspect on a .bin and on a .zip? If the error is reasonable, remove this function. If not, keep it and remove the comment.